### DISCUSSION TITLE
Webhook for SPAM landing

### DISCUSSION THREAD

**Discussion by PrabhuShanmughapriyan** [postalserver/postal#2729] [Feature suggestions]
Hi,
Is there a way in the webhook (or any other mechanism) to find out whether the emails we send have landed in SPAM

**Comment by willpower232** [comment]
Unfortunately you can only get webhooks for actual SMTP events like bounces or delivery problems.

I don't believe there is any way to detect consistently whether emails have ended up in spam as that is all internal to the receiving mail server so kept private from the senders. Microsoft for example just says "Queued mail for delivery" which could be translated as sent to inbox or send to spam.

**Comment by manka86** [comment]
Hi!
I got the point that there is no webhook for landing email in Spam folder. 
But is there a webhook for an event, whet the user make a Spam Complaint or marks an email as Spam?

**Reply by willpower232** [reply]
I don't believe that event is ever emitted outside of the mail server where it happens so it is not possible for Postal to know anything about what happened to the email.

**Reply by manka86** [reply]
then how, for example, sendgrid.com / mailgun.com / mailchimp.com get this info?
I'm trying to figure out how it works in general

**Reply by adamcooke** [reply]
Those providers will subscribe to the feedback loops offered by mailbox providers like Microsoft, Yahoo, Fastmail etc... You would need to subscribe to each FBL by contacting each mailbox provider. Google does things slightly differently and it's not possible to individually identify which emails are marked as spam or individual complaints.

It would be nice if Postal could receive and parse ARF e-mails from FBL providers so it could trigger webhooks in response to spam complaints.

**Reply by manka86** [reply]
Aha, I see. 
Thank you so much!
